[TEST-FAILURE] SlowOperationDetector_EntryProcessorTest failures

com.hazelcast.spi.impl.operationexecutor.slowoperationdetector.SlowOperationDetector_EntryProcessorTest.testSlowEntryProcessor
com.hazelcast.spi.impl.operationexecutor.slowoperationdetector.SlowOperationDetector_EntryProcessorTest.testMultipleSlowEntryProcessorClasses

```
java.lang.AssertionError: Expected 4 total invocations, but was 3 expected:<4> but was:<3>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at com.hazelcast.test.HazelcastTestSupport.assertEqualsStringFormat(HazelcastTestSupport.java:672)
	at 
```
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-sonar/com.hazelcast$hazelcast/600/testReport/junit/com.hazelcast.spi.impl.operationexecutor.slowoperationdetector/SlowOperationDetector_EntryProcessorTest/testSlowEntryProcessor/

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetectorAbstractTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetectorAbstractTest.java
@@ -29,7 +29,6 @@ import com.hazelcast.spi.AbstractOperation;
 import com.hazelcast.spi.Operation;
 import com.hazelcast.spi.impl.operationservice.InternalOperationService;
 import com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl;
-import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.util.EmptyStatement;
 
@@ -47,8 +46,6 @@ import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
 abstract class SlowOperationDetectorAbstractTest extends HazelcastTestSupport {
-
-    static final long ASSERT_TRUE_TIMEOUT = 10;
 
     private static final String DEFAULT_KEY = "key";
     private static final String DEFAULT_VALUE = "value";
@@ -113,22 +110,12 @@ abstract class SlowOperationDetectorAbstractTest extends HazelcastTestSupport {
     }
 
     static void assertNumberOfSlowOperationLogs(final Collection<SlowOperationLog> logs, final int expected) {
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                assertEqualsStringFormat("Expected %d slow operation logs, but was %d.", expected, logs.size());
-            }
-        }, ASSERT_TRUE_TIMEOUT);
+        assertEqualsStringFormat("Expected %d slow operation logs, but was %d.", expected, logs.size());
     }
 
     static void assertTotalInvocations(final SlowOperationLog log, final int totalInvocations) {
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                assertEqualsStringFormat("Expected %d total invocations, but was %d. Log: " + log.createDTO().toJson(),
-                        totalInvocations, log.totalInvocations.get());
-            }
-        }, ASSERT_TRUE_TIMEOUT);
+        assertEqualsStringFormat("Expected %d total invocations, but was %d. Log: " + log.createDTO().toJson(),
+                totalInvocations, log.totalInvocations.get());
     }
 
     static void assertEntryProcessorOperation(SlowOperationLog log) {

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetectorBasicTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetectorBasicTest.java
@@ -100,7 +100,7 @@ public class SlowOperationDetectorBasicTest extends SlowOperationDetectorAbstrac
     public void testSlowOperationOnGenericOperationThread() {
         instance = getSingleNodeCluster(1000);
 
-        SlowOperation operation = new SlowOperation(2);
+        SlowOperation operation = new SlowOperation(3);
         executeOperation(instance, operation);
         operation.await();
 
@@ -117,7 +117,7 @@ public class SlowOperationDetectorBasicTest extends SlowOperationDetectorAbstrac
     public void testSlowOperationOnPartitionOperationThread() {
         instance = getSingleNodeCluster(1000);
 
-        SlowOperation operation = new SlowOperation(4, 1);
+        SlowOperation operation = new SlowOperation(4, 2);
         executeOperation(instance, operation);
         operation.await();
 
@@ -171,13 +171,13 @@ public class SlowOperationDetectorBasicTest extends SlowOperationDetectorAbstrac
     @Test
     public void testSlowRecursiveOperation() {
         int partitionThreads = 32;
-        int numberOfOperations = 50;
-        int recursionDepth = 20;
+        int numberOfOperations = 40;
+        int recursionDepth = 15;
 
         Config config = new Config();
         config.setProperty(GroupProperties.PROP_SLOW_OPERATION_DETECTOR_THRESHOLD_MILLIS, "1000");
+        config.setProperty(GroupProperties.PROP_SLOW_OPERATION_DETECTOR_LOG_RETENTION_SECONDS, String.valueOf(Integer.MAX_VALUE));
         config.setProperty(GroupProperties.PROP_PARTITION_OPERATION_THREAD_COUNT, String.valueOf(partitionThreads));
-
         instance = createHazelcastInstance(config);
 
         List<SlowRecursiveOperation> operations = new ArrayList<SlowRecursiveOperation>(numberOfOperations);
@@ -186,7 +186,7 @@ public class SlowOperationDetectorBasicTest extends SlowOperationDetectorAbstrac
         int partitionIndex = 1;
         for (int i = 0; i < numberOfOperations; i++) {
             int partitionId = partitionIndex % partitionCount;
-            SlowRecursiveOperation operation = new SlowRecursiveOperation(partitionId, recursionDepth, 3);
+            SlowRecursiveOperation operation = new SlowRecursiveOperation(partitionId, recursionDepth, 4);
             operations.add(operation);
             executeOperation(instance, operation);
             partitionIndex += partitionCount / partitionThreads + 1;

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector_EntryProcessorTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector_EntryProcessorTest.java
@@ -20,7 +20,6 @@ import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.IMap;
 import com.hazelcast.map.EntryBackupProcessor;
 import com.hazelcast.map.EntryProcessor;
-import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.annotation.SlowTest;
 import org.junit.Test;
@@ -40,7 +39,7 @@ public class SlowOperationDetector_EntryProcessorTest extends SlowOperationDetec
 
     @Test
     public void testSlowEntryProcessor() throws InterruptedException {
-        HazelcastInstance instance = getSingleNodeCluster(2000);
+        HazelcastInstance instance = getSingleNodeCluster(1000);
         IMap<String, String> map = getMapWithSingleElement(instance);
 
         for (int i = 0; i < 3; i++) {
@@ -60,13 +59,13 @@ public class SlowOperationDetector_EntryProcessorTest extends SlowOperationDetec
         Collection<SlowOperationLog.Invocation> invocations = getInvocations(firstLog);
         assertEqualsStringFormat("Expected %d invocations, but was %d", 4, invocations.size());
         for (SlowOperationLog.Invocation invocation : invocations) {
-            assertInvocationDurationBetween(invocation, 2000, 6500);
+            assertInvocationDurationBetween(invocation, 1000, 6500);
         }
     }
 
     @Test
     public void testMultipleSlowEntryProcessorClasses() throws InterruptedException {
-        HazelcastInstance instance = getSingleNodeCluster(2000);
+        HazelcastInstance instance = getSingleNodeCluster(1000);
         IMap<String, String> map = getMapWithSingleElement(instance);
 
         for (int i = 0; i < 3; i++) {
@@ -82,22 +81,25 @@ public class SlowOperationDetector_EntryProcessorTest extends SlowOperationDetec
         final Collection<SlowOperationLog> logs = getSlowOperationLogs(instance);
         assertNumberOfSlowOperationLogs(logs, 2);
 
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                Iterator<SlowOperationLog> iterator = logs.iterator();
-                SlowOperationLog firstLog = iterator.next();
-                SlowOperationLog secondLog = iterator.next();
+        Iterator<SlowOperationLog> iterator = logs.iterator();
+        SlowOperationLog firstLog = iterator.next();
+        SlowOperationLog secondLog = iterator.next();
+        Collection<SlowOperationLog.Invocation> firstInvocations = getInvocations(firstLog);
+        Collection<SlowOperationLog.Invocation> secondInvocations = getInvocations(secondLog);
 
-                int firstSize = getInvocations(firstLog).size();
-                int secondSize = getInvocations(secondLog).size();
+        int firstSize = firstInvocations.size();
+        int secondSize = secondInvocations.size();
+        assertTrue(format(
+                        "Expected to find 1 and 4 invocations in logs, but was %d and %d. First log: %s%nSecond log: %s",
+                        firstSize, secondSize, firstLog.createDTO().toJson(), secondLog.createDTO().toJson()),
+                (firstSize == 1 ^ secondSize == 1) && (firstSize == 4 ^ secondSize == 4));
 
-                assertTrue(format(
-                                "Expected to find 1 and 4 invocations in logs, but was %d and %d. First log: %s%nSecond log: %s",
-                                firstSize, secondSize, firstLog.createDTO().toJson(), secondLog.createDTO().toJson()),
-                        (firstSize == 1 ^ secondSize == 1) && (firstSize == 4 ^ secondSize == 4));
-            }
-        }, ASSERT_TRUE_TIMEOUT);
+        for (SlowOperationLog.Invocation invocation : firstInvocations) {
+            assertInvocationDurationBetween(invocation, 1000, 5500);
+        }
+        for (SlowOperationLog.Invocation invocation : secondInvocations) {
+            assertInvocationDurationBetween(invocation, 1000, 5500);
+        }
     }
 
     @Test
@@ -105,7 +107,7 @@ public class SlowOperationDetector_EntryProcessorTest extends SlowOperationDetec
         HazelcastInstance instance = getSingleNodeCluster(1000);
         IMap<String, String> map = getMapWithSingleElement(instance);
 
-        NestedSlowEntryProcessor entryProcessor = new NestedSlowEntryProcessor(map, 2);
+        NestedSlowEntryProcessor entryProcessor = new NestedSlowEntryProcessor(map, 3);
         map.executeOnEntries(entryProcessor);
         entryProcessor.await();
 

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector_JsonTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector_JsonTest.java
@@ -98,10 +98,10 @@ public class SlowOperationDetector_JsonTest extends SlowOperationDetectorAbstrac
     @Test
     public void testJSON_SlowEntryProcessor() {
         for (int i = 0; i < 2; i++) {
-            map.executeOnEntries(getSlowEntryProcessor(2));
+            map.executeOnEntries(getSlowEntryProcessor(3));
         }
+        map.executeOnEntries(getSlowEntryProcessor(4));
         map.executeOnEntries(getSlowEntryProcessor(3));
-        map.executeOnEntries(getSlowEntryProcessor(2));
         awaitSlowEntryProcessors();
 
         logger.finest(getOperationStats(instance).toString());
@@ -114,9 +114,9 @@ public class SlowOperationDetector_JsonTest extends SlowOperationDetectorAbstrac
     @Test
     public void testJSON_multipleEntryProcessorClasses() throws InterruptedException {
         for (int i = 0; i < 2; i++) {
-            map.executeOnEntries(getSlowEntryProcessor(2));
+            map.executeOnEntries(getSlowEntryProcessor(3));
         }
-        SlowEntryProcessorChild entryProcessorChild = new SlowEntryProcessorChild(2);
+        SlowEntryProcessorChild entryProcessorChild = new SlowEntryProcessorChild(3);
         map.executeOnEntries(entryProcessorChild);
         map.executeOnEntries(getSlowEntryProcessor(4));
 

